### PR TITLE
VZ-9994 Add optional external cert-manager to dynamic update pipeline

### PR DESCRIPTION
**Pull request by mbenmclean**
This adds external cert-manager as an option to the dynamic update pipeline. It also includes it as a configuration of the dynamic updates suite that runs from the periodics.

### CODE REVIEW

File: ci/dynamic-updates/JenkinsfileTrigger [hunk 1]
```diff
@@ -127,19 +128,36 @@ pipeline {
...
+                                startDCTestJob([booleanParam(name: 'RUN_INFRA_TESTS', value: true),
+                                                booleanParam(name: 'RUN_AVAILABILITY_STATUS', value: true),
                                                 booleanParam(name: 'RUN_AUTHPROXY', value: true),
                                                 booleanParam(name: 'RUN_CERT_MANAGER', value: true),
                                                 booleanParam(name: 'RUN_FLUENTD', value: true)])
                             }
                         }
                     }
                 }
+                stage ('Availability Status, AuthProxy, CertManager, and Fluentd Update Tests With External cert-manager') {
```
Comment (michael-cico) [inline ci/dynamic-updates/JenkinsfileTrigger:140]: Should we run this here, which runs on every periodic, or as part of the weeklies?

File: ci/scripts/prepare_jenkins_at_environment.sh [hunk 2]
```diff
@@ -137,6 +138,10 @@ EOF
     exit 1
   fi
 
+  if [ $INSTALL_EXTERNAL_CERT_MANAGER == true ]; then
```
Comment (michael-cico) [inline ci/scripts/prepare_jenkins_at_environment.sh:141]: not for this pass, but if this is installing CM and NGINX, should it be something like `INSTALL_THIRDPARTY_COMPONENTS` ?

Don't want to be pedantic, but if it's installing more than one thing it should probably be named appropriately.

File: ci/dynamic-updates/JenkinsfileTrigger (hunk 1 above)
Comment (michael-cico) [reply ci/dynamic-updates/JenkinsfileTrigger:140]: I guess we could run it here for now, but move it to the weeklies once we're confident it's stable.

File: tests/e2e/security/netpol/network_policy_test.go [hunk 3]
```diff
@@ -167,9 +169,17 @@ var _ = t.Describe("Test Network Policies", Label("f:security.netpol"), func() {
...
-				t.Logs.Info("Test cert-manager ingress rules")
-				err := testAccess(metav1.LabelSelector{MatchLabels: map[string]string{kubernetesAppLabel: "prometheus"}}, vzconst.PrometheusOperatorNamespace, metav1.LabelSelector{MatchLabels: map[string]string{"app": vzconst.CertManagerNamespace}}, vzconst.CertManagerNamespace, 9402, false, true)
-				Expect(err).To(BeNil(), fmt.Sprintf("FAIL: Test cert-manager ingress failed: reason = %s", err))
+				kubeconfig, err := k8sutil.GetKubeConfigLocation()
+				Expect(err).To(BeNil())
+				vz, err := pkg.GetVerrazzanoInstallResourceInCluster(kubeconfig)
+				if err != nil {
+					Expect(err).To(BeNil())
+				}
+				if vzcr.IsComponentStatusEnabled(vz, certmanager.ComponentName) {
```
Comment (michael-cico) [inline tests/e2e/security/netpol/network_policy_test.go:178]: Is this testing the network access from prometheus to CM for scraping?

File: ci/dynamic-updates/JenkinsfileTrigger (hunk 1 above)
Comment (mbenmclean) [reply ci/dynamic-updates/JenkinsfileTrigger:140]: Probably easier to run here. We don't run this from the weeklies right now. Also doesn't really get checked

File: tests/e2e/verify-infra/vmi/vmi_test.go [hunk 4]
```diff
@@ -406,110 +404,8 @@ var _ = t.Describe("VMI", Label("f:infra-lcm"), func() {
 			})
 		}
 	})
-
```
Comment (michael-cico) [inline tests/e2e/verify-infra/vmi/vmi_test.go:409]: why are these tests removed?

File: ci/scripts/prepare_jenkins_at_environment.sh (hunk 2 above)
Comment (mbenmclean) [reply ci/scripts/prepare_jenkins_at_environment.sh:141]: Ok will do as follow up

File: tests/e2e/security/netpol/network_policy_test.go (hunk 3 above)
Comment (mbenmclean) [reply tests/e2e/security/netpol/network_policy_test.go:178]: It looks like it. I didn't add this test so I'm not entirely sure of the purpose

**Review by michael-cico — APPROVED** [review]

**Pull request merged**